services/src/main/java/org/keycloak/protocol/oid4vc/model/VerifiableCredential.java

```diff
@@ -86,11 +111,30 @@ public VerifiableCredential setType(List<String> type) {
         return this;
     }
 
-    public URI getIssuer() {
+    public Object getIssuer() {
         return issuer;
     }
 
-    public VerifiableCredential setIssuer(URI issuer) {
+    public VerifiableCredential setIssuer(Object issuer) {
+        this.issuer = issuer;
+        if (issuer instanceof Map<?, ?> isserMap) {
+
+            Optional.ofNullable(isserMap).ifPresent(map -> {
+                String id = (String) Optional.ofNullable(map.get("id"))
+                                             .orElseThrow(() -> new IllegalArgumentException(
+                                                     "id is a required field for issuer"));
+                try {
+                    // id must be a URL: https://www.w3.org/TR/vc-data-model-2.0/#issuer
+                    new URI(id);
```

Comment:
This check for valid URI is only executed on issuer expanded property<1>, and not issuer property<2>.
It might be better to check it on both cases. WDYT?

[<1>Usage of issuer expanded property in VCDM spec's example](https://www.w3.org/TR/vc-data-model-2.0/#example-usage-of-issuer-expanded-property)
```
  "issuer": {
    "id": "did:example:76e12ec712ebc6f1c221ebfeb1f",
    "name": "Example University"
  },
```

[<2>Usage of Usage of issuer property in VCDM spec's example](https://www.w3.org/TR/vc-data-model-2.0/#example-usage-of-issuer-property)
```
  "issuer": "https://university.example/issuers/14",
```